### PR TITLE
feat(API): update and delete secret for managing organization secrets

### PR DESCRIPTION
**Pull request by appleboy**
- Add `UpdateSecret` function to modify org or user repo secret
- Add `DeleteSecret` function to delete secret from an organization
- Add `UpdateSecretOption` struct for updating secret options
- Add `UpdateOrgSecret` function to update a secret in an organization
- Add `DeleteOrgSecret` function to delete a secret in an organization

GitHub API

1. Update Org Secret: https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28#create-or-update-an-organization-secret
2. Delete Org Secret: https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28#delete-an-organization-secret

### CODE REVIEW

**Comment by techknowlogick** [comment]
You use `ChangeSecret` and `UpdateSecretOption`, could you align the naming to use either change or update?
Also, how do these APIs compare to GitHub's for the same function?

**Comment by appleboy** [comment]
@techknowlogick 14b25cd for `ChangeSecret and UpdateSecretOption, could you align the naming to use either change or update?`

**Comment by appleboy** [comment]
GitHub API

1. Update Org Secret: https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28#create-or-update-an-organization-secret
2. Delete Org Secret: https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28#delete-an-organization-secret

**Review by lunny — APPROVED** [review]

File: models/secret/secret.go [hunk 1]
```diff
@@ -93,3 +94,48 @@ func FindSecrets(ctx context.Context, opts FindSecretsOptions) ([]*Secret, error
...
 	return db.GetEngine(ctx).Where(opts.toConds()).Count(new(Secret))
 }
+
+// UpdateSecret changes org or user reop secret.
+func UpdateSecret(ctx context.Context, orgID, repoID int64, name, data string) error {
+	sc := new(Secret)
+	has, err := db.GetEngine(ctx).
+		Where("owner_id=?", orgID).
+		And("repo_id=?", repoID).
+		And("name=?", strings.ToUpper(name)).
```
Comment (wxiaoguang) [inline models/secret/secret.go]: Here and below, the `ToUpper` doesn't seem consist with other code , eg: `InsertEncryptedSecret`

File: routers/api/v1/org/action.go [hunk 2]
```diff
@@ -113,3 +113,82 @@ func CreateOrgSecret(ctx *context.APIContext) {
...
+	//   "204":
+	//     description: delete one secret of the organization
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	secretName := ctx.Params(":secretname")
+	err := secret_model.DeleteSecret(
+		ctx, ctx.Org.Organization.ID, 0, secretName,
+	)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "DeleteSecret", err)
```
Comment (wxiaoguang) [inline routers/api/v1/org/action.go]: It doesn't seem right: deleting a non-existing secret cause 500 internal server error

File: routers/api/v1/org/action.go [hunk 3]
```diff
@@ -113,3 +113,82 @@ func CreateOrgSecret(ctx *context.APIContext) {
...
+	//     description: update one secret of the organization
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	secretName := ctx.Params(":secretname")
+	opt := web.GetForm(ctx).(*api.UpdateSecretOption)
+	err := secret_model.UpdateSecret(
+		ctx, ctx.Org.Organization.ID, 0, secretName, opt.Data,
+	)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "ChangeSecret", err)
```
Comment (wxiaoguang) [inline routers/api/v1/org/action.go]: (same as deleting, non-existing secret causes 500 server error)

File: routers/api/v1/api.go [hunk 4]
```diff
@@ -1301,6 +1301,9 @@ func Routes() *web.Route {
 			m.Group("/actions/secrets", func() {
 				m.Get("", reqToken(), reqOrgOwnership(), org.ListActionsSecrets)
 				m.Post("", reqToken(), reqOrgOwnership(), bind(api.CreateSecretOption{}), org.CreateOrgSecret)
```
Comment (wxiaoguang) [inline routers/api/v1/api.go:1303]: There was bug in `CreateOrgSecret`: It doesn't validate the name (see actions.NameRegexMatch)

File: models/secret/secret.go (hunk 1 above)
Comment (appleboy) [reply models/secret/secret.go]: See: https://github.com/go-gitea/gitea/blob/e4b2bdfbc0a28dcb4272de923b547a19131abd22/models/secret/secret.go#L34

File: routers/api/v1/org/action.go (hunk 2 above)
Comment (appleboy) [reply routers/api/v1/org/action.go]: fixed in da6104567

File: routers/api/v1/org/action.go (hunk 3 above)
Comment (appleboy) [reply routers/api/v1/org/action.go]: fixed in da6104567

File: routers/api/v1/api.go (hunk 4 above)
Comment (appleboy) [reply routers/api/v1/api.go:1303]: fixed in 3a4cd0cfe

**Comment by appleboy** [comment]
@wxiaoguang Fixed all that you mention.

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**